python/ruff-ecosystem/ruff_ecosystem/projects.py

```diff
@@ -86,83 +82,21 @@ def as_string():
 
         return hash(as_string())
 
-    @contextlib.contextmanager
-    def patch_config(
-        self,
-        dirpath: Path,
-        preview: bool,
-    ) -> None:
-        """
-        Temporarily patch the Ruff configuration file in the given directory.
-        """
-        dot_ruff_toml = dirpath / ".ruff.toml"
-        ruff_toml = dirpath / "ruff.toml"
-        pyproject_toml = dirpath / "pyproject.toml"
-
-        # Prefer `ruff.toml` over `pyproject.toml`
-        if dot_ruff_toml.exists():
-            path = dot_ruff_toml
-            base = []
-        elif ruff_toml.exists():
-            path = ruff_toml
-            base = []
-        else:
-            path = pyproject_toml
-            base = ["tool", "ruff"]
-
+    def to_ruff_args(self, preview: bool = False) -> list[str]:
         overrides = {
             **ALWAYS_CONFIG_OVERRIDES,
             **self.always,
             **(self.when_preview if preview else self.when_no_preview),
         }
 
-        if not overrides:
-            yield
-            return
-
-        # Read the existing content if the file is present
-        if path.exists():
-            contents = path.read_text()
-            toml = tomli.loads(contents)
-        else:
-            contents = None
-            toml = {}
-
-            # Do not write a toml file if it does not exist and we're just nulling values
-            if all((value is None for value in overrides.values())):
-                yield
-                return
-
-        # Update the TOML, using `.` to descend into nested keys
+        args = []
         for key, value in overrides.items():
-            if value is not None:
-                logger.debug(f"Setting {key}={value!r} in {path}")
-            else:
-                logger.debug(f"Restoring {key} to default in {path}")
-
-            target = toml
-            names = base + key.split(".")
-            for name in names[:-1]:
-                if name not in target:
-                    target[name] = {}
-                target = target[name]
-
             if value is None:
-                # Remove null values i.e. restore to default
-                target.pop(names[-1], None)
+                args.extend(["--config", f"{key} = ''"])
```

Comment:
@AlexWaygood Here I'm trying to unset / restore to default some values.

This is the part of the PR I'm the most unsure about, do you know if that's the correct way to do it ?